PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/action/WhatsNewAction.java

```diff
@@ -0,0 +1,145 @@
+import com.microsoft.azure.toolkit.lib.common.task.AzureTaskManager;
+import org.apache.commons.lang.StringUtils;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.intellij.plugins.markdown.ui.split.SplitFileEditor;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class WhatsNewAction extends AnAction {
+    private static final String WHATSNEW_URL = "https://github.com/microsoft/azure-tools-for-java/tree/develop/PluginsAndFeatures/" +
+            "azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/resources/whatsnew.md";
+    private static final String AZURE_TOOLKIT_FOR_JAVA = "What's New in Azure Toolkit for Java";
+    private static final String AZURE_TOOLKIT_WHATS_NEW = "AzureToolkit.WhatsNew";
+    private static final String VERSION_PATTERN = "<!-- Version: (.*) -->";
+    private static final String WHATS_NEW_CONTENT = "WHAT_S_NEW";
+    private static final String WHAT_S_NEW_CONTENT_PATH = "/whatsnew.md";
+    private static final Key<String> WHATS_NEW_ID = new Key<>(WHATS_NEW_CONTENT);
+    private static final String FAILED_TO_LOAD_WHATS_NEW = "Failed to load what's new document";
+
+    @Override
+    public void actionPerformed(@NotNull final AnActionEvent event) {
+        showWhatsNew(true, event.getProject());
+    }
+
+    @AzureOperation(name = "common.load_whatsnew", type = AzureOperation.Type.ACTION)
+    public static synchronized void showWhatsNew(boolean manually, @NotNull Project project) {
+        final String content = getWhatsNewContent();
+        final DefaultArtifactVersion currentVersion = getVersion(content);
+        final DefaultArtifactVersion lastVersion = getLastVersion();
+        if (manually || compare(currentVersion, lastVersion)) {
+            saveVersion(currentVersion);
+            doShow(content, currentVersion, manually, project);
+        }
+    }
+
+    private static void doShow(String content, DefaultArtifactVersion version, boolean manually, @NotNull Project project) {
+        final FileEditorManager manager = FileEditorManager.getInstance(project);
+        final VirtualFile file = Arrays.stream(manager.getOpenFiles())
+                .filter(f -> StringUtils.equals(f.getUserData(WHATS_NEW_ID), WHATS_NEW_CONTENT))
+                .findFirst().orElse(createVirtualFile(content));
+        AzureTaskManager.getInstance().runAndWait(() -> {
+            if (manager.getProject().isDisposed()) {
+                return;
+            }
+            final FileEditor[] editors = manager.openFile(file, true, true);
+            if (editors.length > 0) { // opened as markdown in editor when markdown plugin enabled.
```

Comment:
What will IntelliJ FileEditor do without markdown plugin? Will it open whats new as plain doc?